src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -109,6 +155,89 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
     });
   }
 
+  get datasetStructures() {
+    return [
+      { value: "AsColumns", label: _t("As Columns") },
+      { value: "AsRows", label: _t("As Rows") },
+    ];
+  }
+
+  setDatasetStructure(value: string) {
+    const switchAxes = value === "AsRows";
+    const oldDataSets = this.props.definition.dataSets;
+    const dataRanges = oldDataSets.map((d) => d.dataRange);
+    const dataSets = this.transposeDataSet(
+      [this.props.definition.labelRange, ...dataRanges],
+      switchAxes
+    );
+    if (dataSets.length === 0) {
+      return;
+    }
+    const labelRange = dataSets.shift()!.dataRange;
+
+    this.props.updateChart(this.props.figureId, {
+      switchAxes,
+      labelRange,
+      dataSets,
+    });
+    this.dataSets = dataSets;
+    this.labelRange = labelRange;
+  }
+
+  private transposeDataSet(
+    dataRanges: (string | undefined)[],
+    switchAxes: boolean | undefined = this.switchAxes
+  ): { dataRange: string }[] {
+    if (switchAxes === undefined) {
+      return dataRanges.filter(isDefined).map((dataRange) => ({ dataRange }));
+    }
+    const zonesBySheetName = {};
+    const transposedDatasets: { dataRange: string }[] = [];
+    const getters = this.env.model.getters;
+    const figureSheetId = getters.getFigureSheetId(this.props.figureId);
+    let name = getters.getActiveSheet().name;
+    if (figureSheetId) {
+      name = getters.getSheet(figureSheetId).name;
+    }
+    for (const dataRange of dataRanges) {
+      if (!dataRange) {
+        continue;
+      }
+      let { sheetName, xc } = splitReference(dataRange);
+      sheetName = sheetName ?? name;
+      if (!zonesBySheetName[sheetName]) {
+        zonesBySheetName[sheetName] = [];
+      }
+      zonesBySheetName[sheetName].push(toZone(xc));
+    }
+    for (const sheetName in zonesBySheetName) {
+      const zones = zonesBySheetName[sheetName];
+      const contiguousZones = mergeContiguousZones(zones);
+      if (!switchAxes) {
+        for (const zone of contiguousZones) {
+          for (let col = zone.left; col <= zone.right; col++) {
+            let newRange = zoneToXc({ top: zone.top, bottom: zone.bottom, left: col, right: col });
+            if (sheetName) {
```

Comment:
always true
I guess the condition should be to check if it's the name of the current active sheet?

Reply:
No, it should not be checked. It was a relicate from the "not so old" time where I didn't change the sheetName if it was undefined (which is now done a few lines above, with `sheetName = sheetName ?? name`)